src/shared/tpm2-util.c

```diff
@@ -4918,6 +4918,131 @@ int tpm2_undefine_policy_nv_index(
         log_debug("Undefined NV index 0x%x", nv_index);
         return 0;
 }
+
+int tpm2_seal_data(
+                Tpm2Context *c,
+                const struct iovec *data,
+                const Tpm2Handle *primary_handle,
+                const Tpm2Handle *encryption_session,
+                const TPM2B_DIGEST *policy,
+                struct iovec *ret_public,
+                struct iovec *ret_private) {
+
+        int r;
+
+        assert(c);
+        assert(data);
+        assert(primary_handle);
+
+        /* This is a generic version of tpm2_seal(), that doesn't imply any policy or any specific
+         * combination of the two keypairs in their marshalling. tpm2_seal() is somewhat specific to the FDE
+         * usecase. We probably should migrate tpm2_seal() to use tpm2_seal_data() eventually. */
+
+        if (data->iov_len >= sizeof_field(TPMS_SENSITIVE_CREATE, data.buffer))
```

Comment:
Actually, tpm2-tss defines the size of this incorrectly; I opened a bug: https://github.com/tpm2-software/tpm2-tss/issues/2694

I added this bit in my open PR (I'm trying to get time to get back to it): https://github.com/systemd/systemd/commit/564ff3480027d7079d9a9229b6e94c1e3e3d19b1#diff-e78fb061a09a598606adc02a47739deea16c1d8495cc2df745c8bc52467702c5R31-R35

Reply:
Ah, interesting. Thankfully we never want to store more than 256bit (i.e. 32byte in the buffer, so the size doesn't matter much to us.

This check here is only really safety so that the memcpy() below doesn't write too much into the buffer.

Hence I think we can leave this as is, and once this is solved in tpm2-tss we should continue to just work.